extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/AddressResolverConfiguration.java

```diff
@@ -26,4 +26,15 @@
     @ConfigItem(defaultValue = "0")
     public int cacheNegativeTimeToLive;
 
+    /**
+     * The maximum number of queries to be sent during a resolution.
+     */
+    @ConfigItem(defaultValue = "4")
+    public int maxQueries;
+
+    /**
+     * The amount of time in milliseconds after which a DNS query is considered to be failed.
+     */
+    @ConfigItem(defaultValue = "5000")
+    public long queryTimeout;
```

Comment:
Ideally, that should be a `Duration` if we want to stay consistent with the rest of Quarkus.

Reply:
Yes, duration would be better.

Reply:
Note that vertx config does not use duration much because it was developed before we had duration support. But new properties should use duration when possible.

Reply:
Do you want me to change the type of the other (existing) properties as well?

Reply:
Ok, I'll change the type of queryTimeout to Duration.

> Asking for a separate PR as if we might backport this one, I will definitely not backport a refactoring.

@gsmet Not sure what this is supposed to mean. Can you elaborate a little?

Reply:
@sophokles73 I think you should do it in 2 steps:
- adjust this pull request to use a `Duration` for the options you added
- create another pull request to make things consistent for the ones that are already there